src/main/java/com/google/sps/servlets/MatchServlet.java

```diff
@@ -49,8 +65,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         //return null for matches so that page redirects to logged out homepage
         response.getWriter().println(gson.toJson(null));
     } else {
-        Collection<User> matches = matchRepository.getMatchesForUser(sessionContext.getLoggedInUser());
-        response.getWriter().println(gson.toJson(matches));
+        // calling analyze request 
+        boolean value = analyzeRequest(request);
+        // if true, means the request was passed
+        if(value) {
+            Collection<User> matches = matchRepository.getMatchesForUser(sessionContext.getLoggedInUser());
+            response.getWriter().println(gson.toJson(matches));
+        }
+        // request did not pass
+        else {
+            System.err.println("Error too many requests, so request couldn't be added");
+            // print out alert
+            System.out.println("<script type=\"text/javascript\">");
+            System.out.println("alert('Too many requests, so request could not be processed');");
+            System.out.println("location='MatchServlet.java';");
+            System.out.println("</script>");
+            // redirect bc it doesn't work
```

Comment:
It doesn't work because L79-82 only prints to the console on the backend. You'd want to run this logic in the frontend in response to the failure.

Alternatively, you could redirect to an error page (redirecting to /index is misleading, as it gives the user the impression that their request was successful). The HTTP protocol defines error code responses (the 404 Page Not found being the most well known). A 429 feels appropriate here (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status).

To implement the redirect to one of these failure pages, check first if servlets  natively support redirecting/returning to an error page. Otherwise, you'll have to write your own (just a simple html page explaining that the server is busy at the moment, to try again later).

Reply:
Oh I see. I'll just make a Error 429 page and redirect to it when it isRequestPassed fails.

Reply:
Done